### PR TITLE
Add pages, routing, and layouts section

### PR DESCRIPTION
**Pull request by danroth27**

### CODE REVIEW

File: docs/architecture/blazor-for-web-forms-developers/pages-routing-layouts.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+        Hello @name
+    }
+</div>
+
+@code {
+    string text;
+    string name;
+
+    void OnClick() {
+        name = text;
```
Comment (twsouthwick) [inline docs/architecture/blazor-for-web-forms-developers/pages-routing-layouts.md]: Don't you need the `NotifyStateChanged` or whatever that method is called?

**Review by twsouthwick — APPROVED** [review]

File: docs/architecture/blazor-for-web-forms-developers/pages-routing-layouts.md [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+In Blazor, each page in the app is a component, typically defined in a *.razor* file, with one or more specified routes. Routing mostly happens client-side without involving a specific server request. The browser first makes a request to the root address of the app. A root `Router` component in the Blazor app then handles intercepting navigation requests and them to the correct component.
+
+Blazor also supports *deep linking*. Deep linking occurs when the browser makes a request to a specific route other than the root of the app. Requests for deep links sent to the server are routed to the Blazor app, which then routes the request client-side to the correct component.
+
+A simple page in ASP.NET Web Forms might contain the following markup:
+
+*Name.aspx*
+
+```aspx
```
Comment (mairaw) [inline docs/architecture/blazor-for-web-forms-developers/pages-routing-layouts.md]:
```suggestion
```aspx-csharp
```

File: docs/architecture/blazor-for-web-forms-developers/pages-routing-layouts.md (hunk 1 above)
Comment (danroth27) [reply docs/architecture/blazor-for-web-forms-developers/pages-routing-layouts.md]: Not from an event handler. We assume the state has probably changed after a UI event occurs.

File: docs/architecture/blazor-for-web-forms-developers/pages-routing-layouts.md [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+<base href="~/" />
+```
+
+## Page layout
+
+Page layout in ASP.NET Web Forms is handled by Master Pages. Master Pages define a template with one or more content placeholders that can then be supplied by individual pages. Master Pages are defined in *.master* files and start with the `<%@ Master %>` directive. The content of the *.master* files is coded as you would an *.aspx* page, but with the addition of `<asp:ContentPlaceHolder>` controls to mark where pages can supply content.
+
+*Site.master*
+
+```aspx
```
Comment (mairaw) [inline docs/architecture/blazor-for-web-forms-developers/pages-routing-layouts.md]:
```suggestion
```aspx-csharp
```

File: docs/architecture/blazor-for-web-forms-developers/pages-routing-layouts.md [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+
+The `Router` component discovers the routable components in the specified `AppAssembly` and in the optionally specified `AdditionalAssemblies`. When the browser navigates, the `Router` intercepts the navigation and renders the contents of its `Found` parameter with the extracted `RouteData` if a route matches the address, otherwise the `Router` renders its `NotFound` parameter.
+
+The `RouteView` component handles rendering the matched component specified by the `RouteData` with its layout if it has one. If the matched component doesn't have a layout, then the optionally specified `DefaultLayout` is used.
+
+The `LayoutView` component renders its child content within the specified layout. We'll look at layouts more in detail later in this chapter.
+
+## Navigation
+
+In ASP.NET Web Forms, you trigger a navigation to a different page by returning a redirect response to the browser. Returning a redirect response isn't typically possible in Blazor. Blazor doesn't use a request-reply model. You can, however, trigger browser navigations directly, just like you can with JavaScript.
```
Comment (scottaddie) [inline docs/architecture/blazor-for-web-forms-developers/pages-routing-layouts.md]: I suggest providing a basic `Response.Redirect(...);` example here for Web Forms. Maybe something like the following from a code-behind:

```csharp
        protected void NavigateButton_Click(object sender, EventArgs e)
        {
            Response.Redirect("Counter");
        }
```

Comment (scottaddie) [reply docs/architecture/blazor-for-web-forms-developers/pages-routing-layouts.md]: @danroth27 I'll take care of this. I'm reviewing this PR right now.

**Review by scottaddie — APPROVED** [review]

**Pull request merged**